core/imt/inc/ROOT/TExecutor.hxx

```diff
@@ -188,18 +188,26 @@ private:
    auto Map(F func, std::initializer_list<T> args, R redfunc, unsigned nChunks) -> std::vector<typename std::result_of<F(T)>::type>;
 
    ROOT::EExecutionPolicy fExecPolicy;
+
+// For code simplification. Fallbacks to TSequentialExecutor (#else cases) are never instantiated.
 #ifdef R__USE_IMT
-   std::unique_ptr<ROOT::TThreadExecutor> fThreadExecutor;
+# define R__EXECUTOR_THREAD ROOT::TThreadExecutor
 #else
-   #define fThreadExecutor fSequentialExecutor
+# define R__EXECUTOR_THREAD ROOT::TSequentialExecutor
 #endif
 #ifndef R__WIN32
-   std::unique_ptr<ROOT::TProcessExecutor> fProcessExecutor;
+# define R__EXECUTOR_PROCESS ROOT::TProcessExecutor
 #else
-   #define fProcessExecutor fSequentialExecutor
+# define R__EXECUTOR_PROCESS ROOT::TSequentialExecutor
```

Comment:
Would it be a bit less confusing if we used `int` or `Unused_t` instead of `TSequentialExecutor`? (where e.g. `using Unused_t = int`).

Reply:
That would certainly make it easier to read